packages/embed/src/overlay/overlay.ts

```diff
@@ -36,13 +36,21 @@ export const createOverlayController = (
 
   element.appendChild(Container)
 
+  const beforeUnloadHandler = (event) => {
+    event.preventDefault()
+    subject.approvalCancelled$.next()
+    return (event.returnValue = true)
```

Comment:
> The HTML specification states that authors should use the [Event.preventDefault()](https://developer.mozilla.org/en-US/docs/Web/API/Event/preventDefault) method instead of using [Event.returnValue](https://developer.mozilla.org/en-US/docs/Web/API/Event/returnValue) to prompt the user. However, this is not yet supported by all browsers.

Reply:
- Could this potentially affect any merchant-written code? (i.e. preventDefault)
- Can we add a test for this functionality?

Reply:
Added a test covering that scenario (well, it's simulated)

Reply:
I also checked with an already existing `beforeunload` listener on plantly, it works fine as long as they don't prevent immediate propagation with `event.stopImmediatePropagation`